lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -661,9 +662,24 @@ def collect_comments(node)
       comments = +""
 
       start_line = node.location.start_line - 1
-      start_line -= 1 unless @comments_by_line.key?(start_line)
+      start_line -= 1 unless comment_exists_at?(start_line)
 
+      append_comment_lines(start_line, comments)
+
+      comments.chomp!
+      comments
+    end
+
+    sig { params(line: Integer).returns(T::Boolean) }
+    def comment_exists_at?(line)
+      @comments_by_line.key?(line) || !@source_lines[line - 1].to_s.strip.empty?
+    end
+
+    sig { params(start_line: Integer, comments: String).void }
+    def append_comment_lines(start_line, comments)
```

Comment:
Do we need to extract this method? Can we move it back to the original and minimize the changes?